server/src/test/java/org/elasticsearch/action/admin/indices/datastream/DeleteDataStreamRequestTests.java

```diff
@@ -62,21 +74,93 @@ public void testValidateRequestWithoutName() {
 
     public void testDeleteDataStream() {
         final String dataStreamName = "my-data-stream";
-        DataStream existingDataStream = new DataStream(dataStreamName, "timestamp", Collections.emptyList());
-        ClusterState cs = ClusterState.builder(new ClusterName("_name"))
-            .metadata(Metadata.builder().dataStreams(Map.of(dataStreamName, existingDataStream)).build()).build();
-        DeleteDataStreamAction.Request req = new DeleteDataStreamAction.Request(dataStreamName);
+        final List<String> otherIndices = List.of("foo", "bar", "baz");
+
+        ClusterState cs = getClusterState(
+            List.of(new Tuple<>(dataStreamName, List.of(
```

Comment:
Maybe use `randomSubSetOf` to have 0-2 indices randomly? I think we still need to also support the case where there are no indices.